### ISSUE TITLE
1.0.7 is broken on Dart 2.1

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/package_resolver#23]
See https://travis-ci.org/dart-lang/json_serializable/builds/497899992

Suggestion:

- publish 1.0.7+1 that reverts the last change and restricts to <=2.1.0
- publish 1.0.7+2 that reverts the revert and requires >=2.1.1

**Comment by dotdotcommadot** [comment]
I suppose this issue just broke all of my tests.... 
I'm on dart 2.2.0-dev.0.0.

```[SEVERE] build_web_compilers|ddc on package:package_resolver/package_resolver.dartdevc.module (cached):
Error compiling dartdevc module:package_resolver|lib/package_resolver.ddc.js

[error] The imported library ''current_isolate_resolver_stub.dart'' can't have a part-of directive. (package:package_resolver/src/package_resolver.dart, line 14, col 8)
[error] The imported library ''utils_isolate_stub.dart'' can't have a part-of directive. (package:package_resolver/src/utils.dart, line 16, col 8)
```

**Comment by oksspo** [comment]
Hi guys, I also have problems but with Dart 2.0.0. There is a log:
`Failed to launch the build script. This is likely due to a misconfigured builder definition. See the generated script at .dart_tool/build/entrypoint/build.dart to find errors.
IsolateSpawnException: Unable to spawn isolate: file:///Users/oksana/.pub-cache/hosted/pub.dartlang.org/package_resolver-1.0.7/lib/src/utils_isolate.dart:7:1: Error: 'Future' expects 0 type arguments.
Future<Uri> resolvePackageUri(Uri packageUri) =>
^
file:///Users/oksana/.pub-cache/hosted/pub.dartlang.org/package_resolver-1.0.7/lib/src/utils.dart:36:26: Error: The argument type 'invalid-type' can't be assigned to the parameter type 'dart.core::Uri'.
Try changing the type of the parameter, or casting the argument to 'dart.core::Uri'.
    return loadConfigMap(await isolate.resolvePackageUri(uri), client: client);
                         ^
file:///Users/oksana/.pub-cache/hosted/pub.dartlang.org/package_resolver-1.0.7/lib/src/utils_isolate.dart:8:13: Error: A value of type 'dart.async::Future<dart.core::Uri>' can't be assigned to a variable of type 'invalid-type'.
Try changing the type of the left hand side, or casting the right hand side to 'invalid-type'.
    Isolate.resolvePackageUri(packageUri);
`
Could it be related to the last changes?

**Comment by natebosch** [comment]
@dotdotcommadot - please `pub upgrade` and ensure you have ~~`build_resolvers`~~ `build_modules` at least  ~~`1.0.3`~~ `1.0.8` where that issue was fixed. If you are unable to to get that version of `build_modules` please file an issue at https://github.com/dart-lang/build/issues

**Comment by natebosch** [comment]
I plan to publish this branch to solve this:

https://github.com/dart-lang/package_resolver/tree/1.0.9-backrelease

**Comment by evanweible-wf** [comment]
@natebosch the latest release of `build_resolvers` is `1.0.3` (which does fix this issue) - I'm guessing that's the version you meant to recommend above?

**Comment by natebosch** [comment]
> I'm guessing that's the version you meant to recommend above?

Thanks - yeah I meant `1.0.3`. Fixed above.

**Comment by natebosch** [comment]
Ugh, I take that back - the _real_ fix was in `build_modules` version `1.0.8`... I don't know why I keep confusing myself.
